crates/solidity/testing/perf/npm/src/benchmarks/common.mts

```diff
@@ -0,0 +1,137 @@
+import fs from "node:fs";
+import path from "node:path";
+
+export const hasGC = typeof global.gc == "function";
+
+export async function runGC() {
+  if (hasGC) {
+    global.gc!();
+    await sleep(100);
+  }
+}
+
+export function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export function checkCI() {
+  if (process.env["CI"] == undefined) {
+    console.error("Must run with CI=true");
+  }
+}
+
+export function round2(n: number): number {
+  return Math.round(n * 100) / 100;
+}
+
+export const verboseOption = "verbose";
+
+const verbose = process.argv.includes(verboseOption);
+
+export function log(what: string) {
+  if (verbose) {
+    console.log(what);
+  }
+}
+
+export class SolidityCompilation {
+  constructor(
+    public compilerVersion: string,
+    public fullyQualifiedName: string,
+  ) {}
+
+  public plainVersion(): string {
+    return this.compilerVersion.split("+")[0];
+  }
+
+  public entrypoint(): string {
+    return this.fullyQualifiedName.substring(0, this.fullyQualifiedName.lastIndexOf(":"));
+  }
+
+  public projectName(): string {
+    return this.fullyQualifiedName.substring(this.fullyQualifiedName.lastIndexOf(":") + 1);
+  }
+}
+
+export class SolidityProject {
+  constructor(
+    public sources: Map<string, string>,
+    public compilation: SolidityCompilation,
+  ) {}
+
+  public static build(jsonFile: string): SolidityProject {
+    const json = JSON.parse(fs.readFileSync(jsonFile, "utf8"));
+    let sources = new Map<string, string>();
+
+    // TODO: we should take other information into account too, in particular, the mappings.
```

Comment:
if that is not required/planned for now, let's use a `NOTE` instead, since `TODO` comments are used for blocking work.